NateGoSearch/NateGoSearchQuery.php

```diff
@@ -234,10 +234,14 @@ public function query($keywords)
 		$misspellings = $this->getPopularReplacements($keywords, $misspellings);
 
 		$keywords = $this->normalizeKeywordsForSearching($keywords);
-		$keywords_hash = sha1($keywords);
+		$unique_hash = sha1($keywords.serialize($this->document_types));
```

Comment:
Do we want the hash to be reproducible? I think if we serialize that array, which contains `SwatDBDataobjects`, we probably won't get the same result on a second run.